### PR TITLE
Move scala Notificaion tests from acceptance testsuite to java test file.

### PR DESCRIPTION
**Pull request by Lojjs**
Move scala Notificaion tests from acceptance testsuite to java test file.

Part of Cypher Comparison Support coverage

### CODE REVIEW

File: community/cypher/cypher/src/test/java/org/neo4j/cypher/internal/javacompat/NotificationAcceptanceTest.java [hunk 1]
```diff
@@ -162,182 +189,534 @@ public void shouldNotifyWhenUsingCreateUniqueWhenCypherVersionIs3_2() throws Exc
...
+            assertNotifications( version + "explain cypher runtime=interpreted match (a)-->(b), (c)-->(d) return *", containsItem( cartesianProductWarning ) );
+        } );
+    }
+
+    @Test
+    public void shouldNotNotifyOnCartesianProductWithoutExplain() throws Exception
+    {
+        Stream.of( "CYPHER 2.3", "CYPHER 3.1", "CYPHER 3.2", "CYPHER 3.3" ).forEach(
+                version -> shouldNotNotifyInStream( version, " match (a)-->(b), (c)-->(d) return *" ) );
     }
```
Comment (sherfert) [inline community/cypher/cypher/src/test/java/org/neo4j/cypher/internal/javacompat/NotificationAcceptanceTest.java:656]: Extracting the warnings to variables makes it much more readable. Nice!

File: community/cypher/cypher/src/test/java/org/neo4j/cypher/internal/javacompat/NotificationAcceptanceTest.java [hunk 2]
```diff
@@ -162,182 +189,534 @@ public void shouldNotifyWhenUsingCreateUniqueWhenCypherVersionIs3_2() throws Exc
...
     {
-        assertNotifications( "EXPLAIN\nMATCH (n)\n//TESTING \nMATCH (n:X)\nreturn n Limit 1",
-                containsItem( notification(
-                        "Neo.ClientNotification.Statement.UnknownLabelWarning",
-                        containsString( "the missing label name is: X)" ),
-                        equalTo( new InputPosition( 38, 4, 10 ) ),
-                        SeverityLevel.WARNING ) ) );
+        assertNotifications( "EXPLAIN\nMATCH (n)\n//TESTING \nMATCH (n:X)\nreturn n Limit 1", containsItem( unknownLabelWarning ) );
+    }
+
```
Comment (sherfert) [inline community/cypher/cypher/src/test/java/org/neo4j/cypher/internal/javacompat/NotificationAcceptanceTest.java:710]: Good find that this hasn't been covered so far.

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala [hunk 3]
```diff
@@ -1,629 +0,0 @@
...
-    result.notifications.map(_.getCode) should not contain "Neo.ClientNotification.Statement.EagerOperatorWarning"
-  }
-
-  test("should not warn for eager without load csv") {
-    val result = innerExecute("EXPLAIN MATCH (a), (b) CREATE (c) RETURN *")
-
-    result should use("Eager")
-    result.notifications.map(_.getCode) should not contain "Neo.ClientNotification.Statement.EagerOperatorWarning"
-  }
-
```
Comment (sherfert) [inline enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala:419]: It seems this test is gone too

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala [hunk 4]
```diff
@@ -1,629 +0,0 @@
...
-  test("warn for unfulfillable index seek when using dynamic property lookup with a single label and IN") {
-    graph.createIndex("Person", "name")
-
-    val result = innerExecute("EXPLAIN MATCH (n:Person) WHERE n['key-' + n.name] IN ['Foo', 'Bar'] RETURN n")
-
-    result.notifications should equal(Set(INDEX_LOOKUP_FOR_DYNAMIC_PROPERTY.notification(graphdb.InputPosition.empty,
-                                                                                         indexSeekOrScan(
-                                                                                           Set("Person").asJava))))
-  }
-
```
Comment (sherfert) [inline enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala:339]: This test seems to be gone. (below)

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala [hunk 5]
```diff
@@ -1,629 +0,0 @@
...
-  test("should not warn when using dynamic property lookup with no labels") {
-    graph.createIndex("Person", "name")
-
-    val result = innerExecute("EXPLAIN MATCH (n) WHERE n['key-' + n.name] = 'value' RETURN n")
-
-    result.notifications shouldBe empty
-  }
-
```
Comment (sherfert) [inline enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala:368]: The next two test queries also do not appear in Java. The tests exist, but have a wrong query. Probably a copy-paste error?

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala [hunk 6]
```diff
@@ -1,629 +0,0 @@
...
-    result.notifications.map(_.getCode) should not contain "Neo.ClientNotification.Statement.NoApplicableIndexWarning"
-  }
-
-  test("should not warn for small label scans with merge combined with load csv") {
-    createLabeledNode("A")
-    val result = innerExecute("EXPLAIN LOAD CSV FROM 'file:///ignore/ignore.csv' AS line MERGE (a:A) RETURN *")
-    result should use("LoadCSV", "AntiConditionalApply")
-    result.notifications.map(_.getCode) should not contain "Neo.ClientNotification.Statement.NoApplicableIndexWarning"
-  }
-
```
Comment (sherfert) [inline enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala:455]: Cannot find this test either

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala [hunk 7]
```diff
@@ -1,629 +0,0 @@
...
-  test("should not warn for missing label on update") {
-
-    //when
-    val result = innerExecute("EXPLAIN CREATE (n:Person)")
-
-    //then
-    result.notifications shouldBe empty
-  }
-
```
Comment (sherfert) [inline enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala:479]: Next two also missing

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala [hunk 8]
```diff
@@ -1,629 +0,0 @@
...
-  test("should warn about unbounded shortest path") {
-    val res = innerExecute("EXPLAIN MATCH p = shortestPath((n)-[*]->(m)) RETURN m")
-
-    res.notifications should contain(
-      UNBOUNDED_SHORTEST_PATH.notification(new graphdb.InputPosition(34, 1, 35)))
-  }
-
```
Comment (sherfert) [inline enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala:520]: This one is gone, too.

**Review by sherfert — DISMISSED** [review]
All in all good to join the two tests in Scala and Java. For a couple tests I have not been able to find the equivalent in Java. Might be intentional or not, but it would be good if you could go over them.

File: community/cypher/cypher/src/test/java/org/neo4j/cypher/internal/javacompat/NotificationAcceptanceTest.java (hunk 1 above)
Comment (Lojjs) [reply community/cypher/cypher/src/test/java/org/neo4j/cypher/internal/javacompat/NotificationAcceptanceTest.java:656]: I think the same 👍

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala (hunk 5 above)
Comment (Lojjs) [reply enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala:368]: Must be, good catch! Have fixed it now.

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala (hunk 3 above)
Comment (Lojjs) [reply enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala:419]: Skipped it due to a very similar test already existed but I guess it doesn't hurt to have both so added now.

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala (hunk 6 above)
Comment (Lojjs) [reply enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala:455]: Added

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala (hunk 7 above)
Comment (Lojjs) [reply enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala:479]: Added

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala (hunk 8 above)
Comment (Lojjs) [reply enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala:520]: Added

File: community/cypher/cypher/src/test/java/org/neo4j/cypher/internal/javacompat/NotificationAcceptanceTest.java (hunk 2 above)
Comment (Lojjs) [reply community/cypher/cypher/src/test/java/org/neo4j/cypher/internal/javacompat/NotificationAcceptanceTest.java:710]: 👍

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala (hunk 4 above)
Comment (Lojjs) [reply enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/NotificationAcceptanceTest.scala:339]: Added

**Pull request merged**